### PR TITLE
Basic Character Movement

### PR DESCRIPTION
**Pull request by Wmbat**
Basic character movement using left/right arrow for movement, Space for jumping and Shift for rolling

### CODE REVIEW

File: Assets/Scripts/Player/InputHandler.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+public class InputHandler
+{
+    private readonly Command _jumpCmd = new JumpCommand();
+    private readonly Command _shootCmd = new ShootCommand();
+    private readonly Command _reloadCmd = new ReloadCommand();
+
+    public Command HandleInput()
+    {
+        if (Input.GetKeyDown(KeyCode.Space)) return _jumpCmd;
+        else if (Input.GetKeyDown(KeyCode.R)) return _reloadCmd;
```
Comment (WillTarte) [inline Assets/Scripts/Player/InputHandler.cs:13]: There should be a case for the ShootCommand (Input.GetMouseButton(0))

**Comment by ClaudiaLapalme** [comment]
Would it be possible to put the character movement in the Level1 scene instead of in its own scene?

File: Assets/Scripts/Player/Controller.cs [hunk 2]
```diff
@@ -6,45 +6,117 @@ namespace Player
...
-        public InputHandler InputHandler { get; private set; }
+        #region Interface Variables
+        
         public Rigidbody2D Rigidbody { get; private set; }
         public Animator Animator { get; private set; }
-
+        
         public bool IsGrounded { get; private set; }
-        public bool IsFacingRight { get; set; }
+        public bool IsFacingRight { get; private set; }
```
Comment (WillTarte) [inline Assets/Scripts/Player/Controller.cs:22]: Can you add a TODO comment for this property: When isFacingRight changes (setter) then we also need to either flip the sprite of the active weapon or change the scale of the active weapon gameobject

**Comment by ClaudiaLapalme** [comment]
I'm getting a `tag 'Ground' is undefined` error that causes the character to always stay in falling state

**Review by WillTarte — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Player/Commands/ReloadCommand.cs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+﻿namespace Player.Commands
+{
+    public class ReloadCommand : Command
+    {
+        public override void Execute(Controller controller)
+        {
+            // TODO: Should call a method similar to `controller.ActiveWeapon.Reload()`
```
Comment (ClaudiaLapalme) [inline Assets/Scripts/Player/Commands/ReloadCommand.cs:7]: Just to be sure - this method will be implemented by Alessandro when he glues the HUD, weapons, and character movement together, right?

**Review by ClaudiaLapalme — CHANGES_REQUESTED** [review]
I can't really test the code until it is combined in the Level1 script or has a ground, but the code itself seems to be pretty good

**Comment by WillTarte** [comment]
For the ground check, implementing something similar to https://youtu.be/c3iEl5AwUF8 would be better IMO

File: Assets/Scripts/Player/Commands/ReloadCommand.cs (hunk 3 above)
Comment (WillTarte) [reply Assets/Scripts/Player/Commands/ReloadCommand.cs:7]: Yes

**Comment by WillTarte** [comment]
A few more things:
- In Level1, can you assign a PhysicsMaterial2D to the Platforms' RigidBody? (child of Grid)
- Can you also make the Player GO a prefab if you havent already?
- You also need to create a new tag ("Ground") and assign it to Platforms
- The Camera doesn't follow the player, pls fix
- 

> For the ground check, implementing something similar to https://youtu.be/c3iEl5AwUF8 would be better IMO

Nvm

**Comment by ClaudiaLapalme** [comment]
- Would it be possible to map to the input types instead of to actual keys? That way AWSD would work
- the character's jumping feels pretty sluggish. Might be a good idea to tweak its gravity and velocity
- can you implement going up and down ladders? Otherwise, the player gets hardlocked in the map  towards the end where they MUST climb a ladder to keep going

**Review by ClaudiaLapalme — APPROVED** [review]

**Review by WillTarte — APPROVED** [review]

**Pull request merged**